src/main/kotlin/com/cognifide/gradle/aem/pkg/tasks/PackageConfig.kt

```diff
@@ -0,0 +1,105 @@
+package com.cognifide.gradle.aem.pkg.tasks
+
+import com.cognifide.gradle.aem.AemDefaultTask
+import com.cognifide.gradle.aem.common.file.ZipFile
+import com.cognifide.gradle.aem.common.instance.Instance
+import com.cognifide.gradle.aem.common.instance.service.pkg.Package
+import com.cognifide.gradle.aem.common.instance.service.repository.Node
+import com.cognifide.gradle.aem.common.utils.shortenClass
+import com.cognifide.gradle.common.utils.Patterns
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+open class PackageConfig : AemDefaultTask() {
+
+    /**
+     * Source instance from which OSGi config will be read.
+     */
+    @Internal
+    val instance = aem.obj.typed<Instance> { convention(aem.obj.provider { aem.anyInstance }) }
+
+    /**
+     * Root repository path used to store temporarily OSGi configs for a serialization time only.
+     */
+    @Internal
+    val rootPath = aem.obj.string { convention("/var/gap/package/config") }
+
+    /**
+     * Target directory in which OSGi configs read will be saved.
+     */
+    @Internal
+    val saveDir = aem.obj.dir { convention(aem.packageOptions.configDir) }
+
+    /**
+     * Unique ID of OSGi config to be saved.
+     */
+    @Internal
+    val pid = aem.obj.string { convention(aem.prop.string("package.config.pid")) }
+
+    @TaskAction
+    fun sync() = instance.get().sync {
```

Comment:
This method is definitely too long, please split it.

Reply:
complexity will be increased, many args to be passed between methods, already considered it... but will think again